src/Accounting/index.ts

```diff
@@ -26,24 +27,26 @@ export const calcLeverage: (quote: number, base: number, price: number) => numbe
  * @param maxLeverage The maximum leverage accounts can trade at. This is specific to the Tracer market
  * @returns the price of the asset where the account is eligible for liquidation
  */
-export const calcLiquidationPrice: (quote: number, base: number, price: number, maxLeverage: number) => number = (
+export const calcLiquidationPrice: (quote: BigNumber, base: BigNumber, price: BigNumber, maxLeverage: BigNumber) => BigNumber = (
     quote,
     base,
     price,
     maxLeverage,
 ) => {
     const borrowed = calcBorrowed(quote, base, price);
-    if (borrowed > 0 || base < 0) { // if the user has a position
-        if (base > 0) { // if the user is long
-            return (maxLeverage * (quote - RYAN_6 * LIQUIDATION_GAS_COST)) / (base - maxLeverage * base)
-        } else if (base < 0) { // if the user is short
+    if (borrowed.gt(0) || base.lt(0)) { // if the user has a position
+        if (base.gt(0)) { // if the user is long
             return (
-                -1 * (quote * maxLeverage - RYAN_6 * LIQUIDATION_GAS_COST * maxLeverage)) / (maxLeverage * base + base
+                (maxLeverage.times(quote.minus(RYAN_6.times(LIQUIDATION_GAS_COST)))).div(base.minus(maxLeverage.times(base)))
+            )
+        } else if (base.lt(0)) { // if the user is short
+            return (
+                (quote.times(maxLeverage).minus(RYAN_6.times(LIQUIDATION_GAS_COST).times(maxLeverage)).div(maxLeverage.times(base).plus(base))).negated() 
```

Comment:
Surely my comment from the previous PR is even more relevant now :fearful:  :fearful:  :fearful: